proposals/function-pointers.md

```diff
@@ -263,62 +272,41 @@ We use `System.Runtime.InteropServices.OutAttribute`, applied as a `modreq` to t
-that need to be touched in order to add a new calling convention. A potential solution is to introduce a new encoding that represents
-the calling convention using [`System.Runtime.InteropServices.CallingConvention`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.callingconvention) enum.
-
-For reference, https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/IR/CallingConv.h has the list of calling conventions
-supported by LLVM. While it is unlikely that .NET will ever need to support all of them, it demonstrates that the space of calling
-conventions is very rich.
+### Metadata Representation of Calling Conventions
+
+The following calling convention specifiers are specially recognized by the compiler:
+
+* `System.Runtime.CompilerServices.CallConvCdecl`
+* `System.Runtime.CompilerServices.CallConvStdcall`
+* `System.Runtime.CompilerServices.CallConvFastcall`
+* `System.Runtime.CompilerServices.CallConvThiscall`
+
+If the identifier specified for the `unmanaged` calling convention binds to one of these types, then the function pointer
+is emitted with the calling convention bit set to the appropriate value, and encodes the calling convention type as the first
+modopt on the return type of the function pointer. For any other `CallConv` type or combination of such types, the compiler
+emits a calling convention bit of Unmanaged (0x9), and encodes all `CallConv` types specified as the first modopts on the return
+type of the function pointer. What calling conventions and combinations are valid is determined by the runtime and platform of
+final execution, and no validation is performed that a valid combination of calling conventions has been specified beyond
+ensuring that all calling convention types come from the right namespace in the core library of the program. If the return
+type of a pointer is by `ref`, then all calling conventions will be encoded after the ref specifier.
+
+When interpreting a function pointer in metadata, all modopts that start with `CallConv` and come from the
+`System.Runtime.CompilerServices` namespace in the core library are considered to be part of the calling convention of the
+function pointer.
+
+### UnmanagedCallersOnlyAttribute
+
+`UnamangedCallersOnlyAttribute` is an attribute used by the CLR to indicate that a method should be called with a specific
+calling convention. Because of this, we introduce the following support for working with the attribute:
+
+* It is an error to directly call a method annotated with this attribute from C#. Users must obtain a function pointer to
+the method and then invoke that pointer.
+* It is an error to apply the attribute to anything other than a static method. The C# compiler will mark any non-static
+methods imported from metadata with this attribute as unsupported by the language.
+* It is an error to have non-blittable types as parameters or the return type of a method marked with the attribute.
```

Comment:
The language doesn't, afaik, have a concept of "blittable" today; only "unmanaged". Is it introducing the former concept or using the latter?

Reply:
I mean to write unmanaged, will fix.

Reply:
Will the same apply to `unmanaged` function pointers or just `UnmanagedCallersOnlyAttribute`?